storage/mysql/log_storage.go

```diff
@@ -18,32 +18,34 @@ import (
 
 const getTreePropertiesSQL string = "SELECT AllowsDuplicateLeaves FROM Trees WHERE TreeId=?"
 const getTreeParametersSQL string = "SELECT ReadOnlyRequests From TreeControl WHERE TreeID=?"
-const selectQueuedLeavesSQL string = `SELECT LeafHash,Payload
+const selectQueuedLeavesSQL string = `SELECT LeafRawHash,Payload
 		 FROM Unsequenced
 		 WHERE TreeID=?
-		 ORDER BY QueueTimestamp DESC,LeafHash ASC LIMIT ?`
-const insertUnsequencedLeafSQL string = `INSERT INTO LeafData(TreeId,LeafHash,TheData)
-		 VALUES(?,?,?) ON DUPLICATE KEY UPDATE LeafHash=LeafHash`
-const insertUnsequencedEntrySQL string = `INSERT INTO Unsequenced(TreeId,LeafHash,MessageId,Payload)
-     VALUES(?,?,?,?)`
-const insertSequencedLeafSQL string = `INSERT INTO SequencedLeafData(TreeId,LeafHash,SequenceNumber)
+		 ORDER BY QueueTimestamp DESC,LeafRawHash ASC LIMIT ?`
+const insertUnsequencedLeafSQL string = `INSERT INTO LeafData(TreeId,LeafRawHash,TheData)
+		 VALUES(?,?,?) ON DUPLICATE KEY UPDATE LeafRawHash=LeafRawHash`
+const insertUnsequencedLeafSQLNoDuplicates string = `INSERT INTO LeafData(TreeId,LeafRawHash,TheData)
 		 VALUES(?,?,?)`
+const insertUnsequencedEntrySQL string = `INSERT INTO Unsequenced(TreeId,LeafRawHash,MessageId,Payload)
+     VALUES(?,?,?,?)`
+const insertSequencedLeafSQL string = `INSERT INTO SequencedLeafData(TreeId,LeafRawHash,LeafHash,SequenceNumber)
```

Comment:
I think `Leaf`is only used by the Log...
Anyway, don't worry about it for now as I think we're going to have to move those around a bit.
Pop a TODO in?